### PR TITLE
PIM-9163: total_fields limit of elasticsearch should be configurable

### PR DESCRIPTION
**Pull request by jmleroux**
Total_fields limit of elasticsearch should be configurable on serenity instances

As per this documentation, this is something that can be needed by some clients:

https://docs.akeneo.com/4.0/maintain_pim/common_issues/index.html#the-limit-of-total-fields-is-reached-in-elasticsearch

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -11,3 +11,4 @@ APP_INDEX_HOSTS=elasticsearch:9200
 APP_PRODUCT_AND_PRODUCT_MODEL_INDEX_NAME=akeneo_pim_product_and_product_model
 MAILER_URL=null://localhost?encryption=tls&auth_mode=login&username=foo&password=bar&sender_address=no-reply@example.com
 AKENEO_PIM_URL=http://localhost:8080
+APP_ELASTICSEARCH_TOTAL_FIELDS_LIMIT=1000
```
Comment (jjanvier) [inline .env]: it was set to 10 000
not 1000

File: config/services/pim_parameters.yml [hunk 2]
```diff
@@ -31,6 +31,10 @@ parameters:
     # Maximum number of resources when updating a list of resources
     api_input_max_resources_number: 100
 
+    # Default value if env var is not defined
+    env(APP_ELASTICSEARCH_TOTAL_FIELDS_LIMIT): 1000
```
Comment (jjanvier) [inline config/services/pim_parameters.yml]: same here

File: config/services/pim_parameters.yml [hunk 3]
```diff
@@ -31,6 +31,10 @@ parameters:
...
+    elasticsearch_total_fields_limit: '%env(APP_ELASTICSEARCH_TOTAL_FIELDS_LIMIT)%'
```
Comment (nmarniesse) [inline config/services/pim_parameters.yml]: The last 2 lines could be replaced by:

```
    elasticsearch_total_fields_limit: '%env(default:10000:APP_ELASTICSEARCH_TOTAL_FIELDS_LIMIT)%'
```

I didn't know that, just found it in https://symfony.com/doc/4.4/configuration/env_var_processors.html

File: .env (hunk 1 above)
Comment (jmleroux) [reply .env]: Do not hesitate to use the "suggestion" github feature for this kind of fix :wink:

File: config/services/pim_parameters.yml (hunk 3 above)
Comment (jmleroux) [reply config/services/pim_parameters.yml]: I tested it, it doesn't work:
> env(default:fallback_param:BAR)
> Retrieves the value of the parameter fallback_param when the BAR env var is not available

It's not a default value, but a default parameter and that's not what we want

Comment (nmarniesse) [reply config/services/pim_parameters.yml]: Ah ok, so it should work like that:

```
    elasticsearch_total_fields_limit_default: 10000 
    elasticsearch_total_fields_limit: '%env(default:elasticsearch_total_fields_limit_default:APP_ELASTICSEARCH_TOTAL_FIELDS_LIMIT)%'
```

You can keep your solution, they are both equal IMO.

File: make-file/test.mk [hunk 4]
```diff
@@ -32,7 +32,7 @@ ifeq ($(CI),true)
 	$(DOCKER_COMPOSE) run -T -u www-data --rm php php vendor/bin/phpspec run --format=junit > var/tests/phpspec/specs.xml
 	.circleci/find_non_executed_phpspec.sh
 else
-	${PHP_RUN} vendor/bin/phpspec run
+	${PHP_RUN} vendor/bin/phpspec run -v
```
Comment (jjanvier) [inline make-file/test.mk]: to revert I suppose

File: config/services/pim_parameters.yml [hunk 5]
```diff
@@ -31,6 +31,10 @@ parameters:
     # Maximum number of resources when updating a list of resources
     api_input_max_resources_number: 100
 
+    # Default value if env var is not defined
+    env(APP_ELASTICSEARCH_TOTAL_FIELDS_LIMIT): 10000
```
Comment (BitOne) [inline config/services/pim_parameters.yml]: Use the .env file to define default values.

**Pull request merged**